File: testdata/bench/kotlin/BenchmarkTest00019.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Safe: cookie value used with prepared statement
suspend fun handler00019(call: ApplicationCall) {
    val role = call.request.cookies["role"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val stmt = conn.prepareStatement("SELECT * FROM users WHERE role = ?")
    stmt.setString(1, role)
    stmt.executeQuery()
    call.respondText("OK")
}

File: testdata/bench/go/BenchmarkTest00019.go
package bench

import (
	"database/sql"
	"net/http"
)

// Parameterized with defer
func Handler00019(w http.ResponseWriter, r *http.Request) {
	param := r.PostFormValue("id")
	db, _ := sql.Open("sqlite3", ":memory:")
	stmt, _ := db.Prepare("DELETE FROM temp WHERE id = ?")
	defer stmt.Exec(param)
}
